### PR TITLE
Biggest team blowout mr

### PR DESCRIPTION
**Pull request by mronauli**
add biggest_team_blowout and worst_loss method

### CODE REVIEW

File: lib/game_teams_collection.rb [hunk 1]
```diff
@@ -55,23 +55,9 @@ def away_games_only
...
-
-  def losing_game_ids(team_id)
-    game_ids = []
-    @game_teams_array.each do |game_team|
-      if game_team.result == "LOSS" && game_team.team_id == team_id
-        game_ids << game_team.game_id
-      end
+  def game_ids_by_result(team_id, result)
+    @game_teams_by_id[team_id].map do |game_team|
+        game_team.game_id if game_team.result == result
```
Comment (mronauli) [inline lib/game_teams_collection.rb:60]: wow! what a more efficient way to go about this method

File: test/season_stats_test.rb [hunk 2]
```diff
@@ -26,13 +26,9 @@ def test_it_can_report_head_to_head_results
...
-
-  def test_can_get_difference_between_wins
-    assert_equal [1, 1, 1, 2], @season_stats.difference_between_wins(24)
-  end
-
-  def test_can_get_difference_between_losses
-    assert_equal [1, 1, 1, 2], @season_stats.difference_between_losses(20)
+  
+  def test_can_get_game_score_differentials
+    assert_equal [1, 1, 1, 2], @season_stats.game_score_differentials(20, "LOSS")
```
Comment (mronauli) [inline test/season_stats_test.rb]: thanks @d-atkins

**Review by d-atkins — APPROVED** [review]
I consolidated a lot of methods into one and made them take a result ("WIN" or "LOSE") as a parameter. Currently passing all tests. New methods passing spec harness.

File: lib/game_teams_stats.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@ def total_wins_per_team(team_id)
   end
 
   def average_win_percentage(team_id)
-    total_wins_per_team(team_id.to_i) / @game_teams_collection.total_games_per_team(team_id.to_i).to_f
+    (total_wins_per_team(team_id.to_i) / @game_teams_collection.total_games_per_team(team_id.to_i).to_f).round(2)
```
Comment (tylertomlinson) [inline lib/game_teams_stats.rb:15]: Was planning on moving from Stat Tracker to here. Nice

File: lib/season_stats.rb [hunk 4]
```diff
@@ -28,15 +30,26 @@ def head_to_head_ids(team_id)
...
-      @game_teams_collection.each {|game_team| acc << game_team if game_team.game_id == game.game_id}
-      acc
+  def game_score_differentials(team_id, result)
+    games = @game_teams_collection.game_ids_by_result(team_id, result)
+    games.map do |game_id|
+      @game_collection.games.find{|game| game.game_id == game_id}.difference_between_score
     end
   end
 
+  def biggest_team_blowout(team_id)
```
Comment (tylertomlinson) [inline lib/season_stats.rb:40]: Nice method

File: test/season_stats_test.rb [hunk 5]
```diff
@@ -11,7 +11,6 @@ def setup
   end
 
   def test_it_exists
-    skip
```
Comment (tylertomlinson) [inline test/season_stats_test.rb]: good catch

File: lib/game_teams_collection.rb [hunk 6]
```diff
@@ -55,23 +55,9 @@ def away_games_only
...
-    end
-    game_ids
-  end
-
-  def losing_game_ids(team_id)
-    game_ids = []
-    @game_teams_array.each do |game_team|
-      if game_team.result == "LOSS" && game_team.team_id == team_id
-        game_ids << game_team.game_id
-      end
```
Comment (Yetidancer) [inline lib/game_teams_collection.rb]: love to see the efficiency

**Pull request merged**